modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -4288,30 +4318,33 @@ else if (StringUtil.equals(
 		}
 
 		if (objectField.getListTypeDefinitionId() != 0) {
-			ListTypeEntry listTypeEntry = null;
+			Serializable value = entry.getValue();
 
-			String value = String.valueOf(values.get(entry.getKey()));
+			if (objectField.compareBusinessType(
+					ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {
 
-			for (ListTypeEntry curListTypeEntry :
-					_listTypeEntryLocalService.getListTypeEntries(
-						objectField.getListTypeDefinitionId())) {
+				Iterable<Object> listTypeEntryKeys = _getIterable(value);
 
-				if (Objects.equals(value, curListTypeEntry.getKey())) {
-					listTypeEntry = curListTypeEntry;
+				Iterator<Object> iterator = listTypeEntryKeys.iterator();
 
-					break;
+				if (!iterator.hasNext() && objectField.isRequired()) {
+					throw new ObjectEntryValuesException.Required(
+						objectField.getName());
 				}
```

Comment:
Hey @carolmariaabb. I also added these lines, because I noticed that we were accepting empty array (through the API) in a required field.